contracts/adapter/ERC20Adapter02.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.10;
+pragma abicoder v1;
+
+import '../token/IERC20.sol';
+import '../token/IWETH.sol';
+
+/// @title Brink ERC20Adapter02
+/// @notice Deployed once and used by Brink executors to fulfill swaps.
+contract ERC20Adapter02 {
+  IWETH public weth;
+  bool public initialized;
+
+  /// @dev Max uint
+  uint256 MAX_INT = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+
+  /// @dev Adapter Owner
+  address payable ADAPTER_OWNER = payable(0x71795b2d53Ffbe5b1805FE725538E4f8fBD29e26);
+
+  /// @dev Ethereum address representations
+  IERC20 private constant _ETH_ADDRESS = IERC20(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+  IERC20 private constant _ZERO_ADDRESS = IERC20(0x0000000000000000000000000000000000000000);
+
+  error NotEnoughETH();
+  error NotEnoughTokenIn(uint amount);
+  error NotEnoughTokenOut(uint amount);
+
+  /// @dev initialize the contract with WETH address
+  /// @param _weth Address of weth
+  function initialize (IWETH _weth) external {
+    require(!initialized, 'INITIALIZED');
+    initialized = true;
+    weth = _weth;
+  }
+
+  /// @dev Makes a call to a router with swap byte data
+  /// @dev returns the requested tokenOutAmount to Account and keeps the rest
+  /// @param to address usually a router contract
+  /// @param data swap byte data 
+  /// @param tokenIn Address of the token to be swapped
+  /// @param tokenOut Address of the token to be returned from the swap
+  /// @param tokenOutAmount Amount of tokenOut to transfer to account
+  /// @param account Address of the account to receive the tokenOut
+  /// @param minTokenInArb Minimum amount of tokenIn arbitrage revenue remaining for ADAPTER_OWNER
+  /// @param minTokenOutArb Minimum amount of tokenOut arbitrage revenue remaining for ADAPTER_OWNER
+  function erc20Swap(address to, bytes memory data, IERC20 tokenIn, IERC20 tokenOut, uint tokenOutAmount, address payable account, uint minTokenInArb, uint minTokenOutArb) external payable {
+    if (isETH(tokenIn)) {
```

Comment:
@Kyrrui can you move this to private fn as well? it's repeated in both functions